### ISSUE TITLE
Add option to pass in additional configmap parameters

### ISSUE THREAD

**Issue by nicl-dev** [argoproj/argo-helm#2462]
## TL;DR:

For bootstrapping ArgoCD it would be benefitial to get rid of the extra step of editing the `argocd-cm` ConfigMap manually [to enable kustomize with helm](https://argo-cd.readthedocs.io/en/stable/user-guide/kustomize/#kustomizing-helm-charts). For example, an additional value like `configs.cm.additionalParameters` could help. Alternatively, for this exact problem, `configs.cm.enableKustomizeHelm` would be enough.

## My use case:

Today I tried bootstrapping ArgoCD with Pulumi using the `argo-cd` Chart together with `kustomize` to fully automate my deployment. Right now it looks like this:

```python
import pulumi
import pulumi_kubernetes as kubernetes

config = pulumi.Config()
app_labels = {
    "app": "argocd",
}

# Use Helm to install ArgoCD
argocd = kubernetes.helm.v3.Release(
    "argocd",
    chart="argo-cd",
    name="argo-cd",
    namespace="argocd",
    create_namespace=True,
    repository_opts=kubernetes.helm.v3.RepositoryOptsArgs(
        repo="https://argoproj.github.io/argo-helm",
    ),
    values={
        "server": {
            "service": {
                "type": "NodePort"
            },
        },
        "configs": {
            "repositories": {
                "fellowship-apps": {
                    "url": "https://github.com/nicl-dev/my-repo.git",
                    "password": "token",
                    "username": "placeholder"
                    }
            },
        },
    }
)
``` 

and my `kustomization.yaml`

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
helmCharts:
  - name: argo-cd
    includeCRDs: true
    releaseName: argo-cd
    namespace: argocd
    version: 5.52.0
    repo: "https://argoproj.github.io/argo-helm"
    valuesFile: values.yaml
patches:
  - target:
      kind: ConfigMap
      name: argocd-cm
    patch: |-
      - op: add
        path: /data
        value: 
          kustomize.buildOptions: --enable-helm
```

My plan was to add the `argocd-apps` Chart to the `Pulumi.py` which would then add the ArgoCD app-of-apps that makes use of the `kustomization.yaml`. Unfortunately I didn't make it that far because of this issue and patching the ConfigMap with kustomize before setting `kustomize.buildOptions: --enable-helm` is obviously not possible. Chicken and egg..
 

### Related helm chart

argo-cd

### Describe the solution you'd like

I think a value like `configs.cm.additionalParameters` would be enough to get rid of most problems caused by a missing option to add parameters to the ConfigMap. Alternatively, for this exact problem, `configs.cm.enableKustomizeHelm` would be enough.

If this approach is something you agree with I'd be glad to help implementing this.

**Comment by pdrastil** [comment]
Hi @nicl-dev, what options are missing? The `configs.cm` values allows you to define any arbitrary key that we don't have in chart like this:

```yaml
configs:
  cm:
    # Part of the chart (default value)
    server.insecure: true
    # Not-in the chart but will be rendered in CM
    kustomize.buildOptions: --enable-helm
```

**Comment by nicl-dev** [comment]
Oh wow, I honestly didn't realize that this is already possible. Apologies for the confusion, no options are missing!
